chapel-lang/chapel
# Remove support for the previosuly deprecated single variables

This PR removes support for `single` variables, which have been deprecated for several releases.

Changes done:
- Remove the `_singlevar` wrapper record code
- Remove implementation of `single` variables in the runtime
- Remove parser support for `single`
- Remove all mentions/implementation details of single in the compiler

Testing
- Build and `make check` with/without comm on MacOS
- Build and `make check` with/without comm on Linux
- Full paratest with/without comm

[Reviewed by @]
